src/Serializer/AbstractItemNormalizer.php

```diff
@@ -86,7 +96,7 @@ public function normalize($object, $format = null, array $context = [])
         $context = $this->initContext($resourceClass, $context);
         $context['api_normalize'] = true;
 
-        if (isset($context['resources'])) {
+        if (isset($context['resources']) && $this->hasIri($object)) {
             $resource = $context['iri'] ?? $this->iriConverter->getIriFromItem($object);
```

Comment:
No. `$context['iri']` is only set from the JsonLd `ItemNormalizer` (I think). I don't know what's the reasoning behind, but I did not change it...